core/src/main/java/org/elasticsearch/gateway/DanglingIndicesState.java

```diff
@@ -108,37 +124,77 @@ void cleanupAllocatedDangledIndices(MetaData metaData) {
-     * as dangled already.
+     * as dangled already.  The second tuple value contains dangling indices that should not
+     * be imported but rather should be deleted and cleaned up.
      */
-    Map<Index, IndexMetaData> findNewDanglingIndices(MetaData metaData) {
+    Tuple<Map<Index, IndexMetaData>, Map<Index, IndexMetaData>> findNewDanglingIndices(MetaData metaData) {
         final Set<String> excludeIndexPathIds = new HashSet<>(metaData.indices().size() + danglingIndices.size());
         for (ObjectCursor<IndexMetaData> cursor : metaData.indices().values()) {
             excludeIndexPathIds.add(cursor.value.getIndex().getUUID());
         }
         excludeIndexPathIds.addAll(danglingIndices.keySet().stream().map(Index::getUUID).collect(Collectors.toList()));
         try {
-            final List<IndexMetaData> indexMetaDataList = metaStateService.loadIndicesStates(excludeIndexPathIds::contains);
+            final List<PersistedIndexMetaData> indexMetaDataList = metaStateService.loadIndicesStates(excludeIndexPathIds::contains);
             Map<Index, IndexMetaData> newIndices = new HashMap<>(indexMetaDataList.size());
-            for (IndexMetaData indexMetaData : indexMetaDataList) {
-                if (metaData.hasIndex(indexMetaData.getIndex().getName())) {
-                    logger.warn("[{}] can not be imported as a dangling index, as index with same name already exists in cluster metadata",
-                        indexMetaData.getIndex());
-                } else {
-                    logger.info("[{}] dangling index, exists on local file system, but not in cluster metadata, auto import to cluster state",
-                        indexMetaData.getIndex());
-                    newIndices.put(indexMetaData.getIndex(), indexMetaData);
+            Map<Index, IndexMetaData> delIndices = new HashMap<>();
+            for (PersistedIndexMetaData persistedIndex : indexMetaDataList) {
+                if (persistedIndex != null) {
+                    final Index index = persistedIndex.getIndexMetaData().getIndex();
+                    final String clusterUUID = metaData.clusterUUID();
+                    if (persistedIndex.isSameClusterUUID(clusterUUID)) {
+                        // the cluster metadata's cluster UUID is the same as the cluster UUID that created this index,
+                        // so the index shouldn't be imported as dangling but rather deleted and cleaned up, because the
+                        // dangling files are probably due to remnants from deleted indices (e.g. the node was offline while
+                        // the index was deleted) or failed cleanup
+                        logger.info("[{}] dangling index exists on the local file system, but it is deleted in the metadata of the " +
+                                    "same cluster that created it, deleting the index on the local file system", index.getName());
```

Comment:
can we log the full index object, includeing the index uuid? also it would be good to have the current cluster uuid and the one in the index meta data state.